core/src/main/java/com/datastax/oss/driver/internal/core/metrics/MetricsFactory.java

```diff
@@ -16,8 +16,12 @@
 package com.datastax.oss.driver.internal.core.metrics;
 
 import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metrics.Metrics;
+import java.util.Optional;
 
-public interface MetricUpdaterFactory {
+public interface MetricsFactory {
```

Comment:
Slightly better imo:

```
public interface MetricsFactory<T extends Metrics> {

  Optional<? extends T> getMetrics();
```

Reply:
Based on my reply to your other comment, I think we should avoid specialized `Metrics` subinterfaces, so this wouldn't apply.

Reply:
Sorry, I don't get it. I'm just saying that `Optional<? extends Metrics> getMetrics()` is more correct than `Optional<Metrics> getMetrics()`.

Reply:
The only goal of this method is to be exposed as `Session.getMetrics()`. In what scenario would `? extends Metric` be more useful for a client application?

Reply:
If someone extends `Session` and `Metrics`, then if this method has the return type `Optional<? extends Metrics>`  it would be possible to use covariance and override it with a more specialized `Optional<MySpecialMetrics>`. That is not possible if the original return type is `Optional<Metrics>`. We had a similar problem with `AsyncResultSet.fetchNextPage()` recently.

Reply:
OK that makes sense, I will change it.